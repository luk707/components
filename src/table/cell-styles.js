export default ({ theme }) => ({
  whiteSpace: "nowrap",
  color: theme.color_black,
  fontWeight: 400,
  textAlign: "left",
  padding: "8px 16px"
});
