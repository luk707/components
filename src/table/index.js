export { default as Table } from "./table";
export { default as Th } from "./th";
export { default as Td } from "./td";
export { default as Tr } from "./tr";
export { default as THead } from "./thead";
