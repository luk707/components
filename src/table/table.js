import React from "react";
import styled from "react-emotion";

const TableElement = styled("table")({
  tableLayout: "fixed",
  borderCollapse: "collapse"
});

const TableWrapper = styled("div")(({ theme }) => ({
  overflowX: "auto",
  backgroundColor: theme.color_offWhite
}));

export default props => (
  <TableWrapper>
    <TableElement {...props} />
  </TableWrapper>
);
