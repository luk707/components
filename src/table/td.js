import styled from "react-emotion";
import cellStyles from "./cell-styles";

export default styled("td")(cellStyles);
