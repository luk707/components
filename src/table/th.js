import styled from "react-emotion";
import cellStyles from "./cell-styles";

export default styled("th")(cellStyles, { fontWeight: 700 });
