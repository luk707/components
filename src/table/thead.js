import styled from "react-emotion";
import Tr from "./tr";

export default styled("tbody")({});
