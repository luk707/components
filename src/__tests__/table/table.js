import React from "react";
import { ThemeProvider } from "emotion-theming";
import renderer from "react-test-renderer";

import createTheme from "../../theme/create-theme";
import { Table, THead, Th, Td, Tr } from "../../table";

const theme = createTheme();

describe("Table", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Table>
            <THead>
              <Tr>
                <Th>Firstname</Th>
                <Th>Lastname</Th>
                <Th>Age</Th>
                <Th>Gender</Th>
              </Tr>
            </THead>
            <tbody>
              <Tr>
                <Td>John</Td>
                <Td>Doe</Td>
                <Td>30</Td>
                <Td>Male</Td>
              </Tr>
              <Tr>
                <Td>Jane</Td>
                <Td>Doe</Td>
                <Td>29</Td>
                <Td>Female</Td>
              </Tr>
            </tbody>
          </Table>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
