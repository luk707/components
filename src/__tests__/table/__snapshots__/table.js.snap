// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<div
  className="css-1todvj7"
>
  <table
    className="css-1mid9s7"
  >
    <tbody
      className="css-0"
    >
      <tr
        className="css-0"
      >
        <th
          className="css-q3icmu"
        >
          Firstname
        </th>
        <th
          className="css-q3icmu"
        >
          Lastname
        </th>
        <th
          className="css-q3icmu"
        >
          Age
        </th>
        <th
          className="css-q3icmu"
        >
          Gender
        </th>
      </tr>
    </tbody>
    <tbody>
      <tr
        className="css-0"
      >
        <td
          className="css-1pbxru0"
        >
          John
        </td>
        <td
          className="css-1pbxru0"
        >
          Doe
        </td>
        <td
          className="css-1pbxru0"
        >
          30
        </td>
        <td
          className="css-1pbxru0"
        >
          Male
        </td>
      </tr>
      <tr
        className="css-0"
      >
        <td
          className="css-1pbxru0"
        >
          Jane
        </td>
        <td
          className="css-1pbxru0"
        >
          Doe
        </td>
        <td
          className="css-1pbxru0"
        >
          29
        </td>
        <td
          className="css-1pbxru0"
        >
          Female
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
